### PR TITLE
electron@19

### PR DESCRIPTION
**Pull request by ShaMan123**
See commits for details

Removing electron-settings means loss of the user's local settings file unless we know where to look for it, and we do. I removed it because it didn't play nicely with the new remote IIRC.

### CODE REVIEW

**Comment by abebeos** [comment]
( tried back then with electron 19, not sure why I dropped the attempt)

I think users will yell if they loose their settings - you need to find a way to keep the existent functionality unbroken.

and remember, that as a general rule, existent tests must pass.

**Comment by abebeos** [comment]
looks like node 14 does not work anymore (https://github.com/deepnest-io/Deepnest/actions/runs/12188649407/job/34002112305?pr=159#step:6:10), so test run needs to be adjusted.

**Comment by ShaMan123** [comment]
On it.
Sticking to https://electron-settings.js.org/index.html#configure for backward compat settings

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,32 +12,36 @@ jobs:
   build:
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x, 20.x]
```
Comment (abebeos) [inline .github/workflows/build.yml:15]: you'd just remove 14 here.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -12,32 +12,36 @@ jobs:
...
-        os: [windows-2022, ubuntu-22.04]
+        os: [windows-2022, ubuntu-24.04]
     runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v3
```
Comment (abebeos) [inline .github/workflows/build.yml:19]: avoid changes like this. See, me currently close-to-sleep, superdizzy. Possibly an autoformatter did it?

File: .gitignore [hunk 3]
```diff
@@ -8,7 +8,6 @@
 /bin
 /minkowski
 /deepnest-*win32-x64
-package-lock.json
```
Comment (abebeos) [inline .gitignore:11]: worked till today. such a change should not happen without a reason + a comment somewhere to clarify.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -12,7 +12,7 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [windows-2022, ubuntu-24.04]
+        os: [windows-2022, ubuntu-24.04, macos-14]
```
Comment (abebeos) [inline .github/workflows/build.yml:15]: You MUST avoid this, as a general (not only here in this project) rule:

You should not introduce/change stuff on-the-fly, when doing a specific PR for specific functionality (here: electron 19 update).

Me off machinge now. have fun!

**Comment by abebeos** [comment]
(hey, macos builds...)

File: .github/workflows/build.yml (hunk 4 above)
Comment (ShaMan123) [reply .github/workflows/build.yml]: rm

**Comment by ShaMan123** [comment]
> (hey, macos builds...)

It does on my macos but on ci it builds a corrupted app that macos refuses to open.

File: .gitignore (hunk 3 above)
Comment (ShaMan123) [reply .gitignore:11]: This is the same as your comment against using ubuntu-latest. Without package.json you are trusting the ci machine and npm/node to do an exact install of node modules without actually specifying that is what you want. Extremely undesirable. That is the whole reason of the cmd `npm ci` - install in accordance to lock file on ci.

File: .github/workflows/build.yml (hunk 2 above)
Comment (ShaMan123) [reply .github/workflows/build.yml:19]: This is a git checkout action. I don't see a problem with the bump. If you insist I will downgrade

**Review by ShaMan123 — COMMENTED** [review]
I am done

File: .github/workflows/build.yml (hunk 2 above)
Comment (ShaMan123) [reply .github/workflows/build.yml:19]: This is a benefit (enhanced caching) of using newer versions:
```bash

Run actions/setup-node@v4
Found in cache @ /opt/hostedtoolcache/node/20.18.1/x6[4](https://github.com/deepnest-io/Deepnest/actions/runs/12189545706/job/34004910615?pr=159#step:4:4)
Environment details


```

**Comment by ShaMan123** [comment]
Node is working pretty reliably, see this server impl:
https://github.com/user-attachments/assets/08fa02f6-ff7e-439b-a0c3-84c3b51db033

File: .gitignore (hunk 3 above)
Comment (abebeos) [reply .gitignore:11]: This change is NOT related to this PR (electron@19).

don't touch anything(!) else, don't let your IDE touch anything else (even if 'correct' reformatiing).

I have near-zero energy for reviews. Let it be minimal necessary.

**Comment by abebeos** [comment]
> see this server impl:

what is this? Is this related to 'electron@19' upgrade?

**Comment by abebeos** [comment]
OT: last year whe some budgets were available, I wasted most time on 'replaying' chaotic work within different repos.

if you want to continue work, you MUST stick to the rule to never introduce changes/additions not relevant to the current topic/PR. Even whitespace-plumbing -> do not do it, keep changes at minimum, so todays (and future) reviewers can validate easily.

**Comment by abebeos** [comment]
expressed differently:

Ask yourself "what is necessary to achieve electron19". Do exactly that, thus a reviewer sees only this in the PR.

3a6818e7be19046d045e1e780da4e017c68aaffa is mostly not necessary, even the ubuntu bumb.

**Comment by ShaMan123** [comment]
I get you and can live with most of your points.
Working in whitespace mayhem is very difficult but I can swallow it. 
I have done my best to keep it very minimal.
However, my time is limited as well. 
I decided to push back to upstream for the benefit of the repo since it seemed on the verge of abandonment and that is a sheer shame for such a powerful algorithm.
Mind you that electron is at v33 and esm is way past acceptance in maintenance, making this project extremely outdated, regardless of overall code quality.

The ci commit you referenced is something I pushed in accordance to your review. Ci was failing without me doing anything and you asked to fix it. This minimal change allows you to revive the pipeline and have some confidence when merging the pr, or else you need to build locally to see nothing is broken.

I am not attached to this work. I needed this great algorithm and therefore wanted to contribute to keep it alive. If you can't find the middle ground that is fine, close the pr. I will complete my tasks regardless and will maintain my own fork, with no better alternative.
I have been on both sides of this interaction, as a maintainer with no energy in a project that is alive due to my efforts and as a maintainer with a lot of energy in a dying team. I do not wish to return to neither.
This is it then. Merge the pr, push to it or close it. It is in your hands now.

OT: The video is my goal for this repo. I don't understand why it was bundled into an electron app of all things. Such powerful algorithms should be standalone and have bindings for different platforms.
I wanted to do that exactly and eventually expose the backward compat electron app binding alongside an npm package and maybe if I had time and energy try to use wasm to make it run purely in the browser.

My passion here is strictly an outcome of my appreciation of this great algorithm and vs the state of code, no hard feelings.

**Comment by abebeos** [comment]
> I get you and can live with most of your points. Working in whitespace mayhem is very difficult but I can swallow it. I have done [...]

that's a large one (note that I did not read fully)

PR review ping-pong is very annoying, for both sides. But here, you do very basic mistakes, which you simply shouldn't do.

1 PR: electron19
1 PR: build/test-system

absolutely nothing special.

(I may revise the PR myself those days, simply by deleting the non-relevant stuff. Tragedy that this nice work goes nirvana for tiny sloppy details)

**Comment by abebeos** [comment]
@ShaMan123 , see #160 for an updated PR. I just removed your off-topic build-system tinkering, no idea where this stubberness of yours comes from.

You could now proceed to a follow-up PR where you modify/extend the build/test system (one commit per topic). You need to document(!), e.g. things like that:

https://github.com/deepnest-io/Deepnest/pull/159/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R46-R51

If you stick to this standard-procedure, then I can give you direct access, so you can save time.

**Comment by abebeos** [comment]
superseded by #160

**Pull request closed without merging**